C2DExamples.md

```diff
@@ -0,0 +1,605 @@
+# Compute-to-Data (C2D) Code Examples
+
+Here are the steps:
+
+0. [Prerequisites](#-Prerequisites)
+1. [Initialize services](#-initialize-services)
+2. [Create a new node.js project](#-create-a-new-node.js-project)
+3. [Install dependencies](#-install-dependencies)
+4. [Import dependencies and add variables and constants](#-import-dependencies-and-add-variables-and-constants)
+5. [Initialize accounts and deploy contracts](#-initialize-accounts-and-deploy-contracts)
+6. [Publish a dataset (Data NFT and Datatoken)](#-publish-a-dataset-data-nft-and-datatoken)
+7. [Publish an algorithm (Data NFT and Datatoken)](#-publish-an-algorithm-data-nft-and-datatoken)
+8. [Resolve published datasets and algorithms](#-resolve-published-datasets-and-algorithms)
+9. [Send datatokens to consumer](#-send-datatokens-to-consumer)
+10. [Consumer starts a compute job using a free C2D environment](#-consumer-starts-a-compute-job-using-a-free-c2D-environment)
+10. [Check compute status and get download compute results url](#-check-compute-status-and-get-download-compute-results-url)
+
+Let's go through each step.
+
+## 0. Prerequisites
+Before we start it is important that you have all of the necessary prerequisites installed on your computer.
+- **A Unix based operating system (Linux or Mac)**. If you are a Windows user you can try to run linux inside a virtual machine but this is outside of the scope of this article.
+- **Git**. Instructions for installing Git can be found here: https://git-scm.com/book/en/v2/Getting-Started-Installing-Git
+- **Node.js** can be downloaded from here: https://nodejs.org/en/download/
+- **Docker** can be installed from here: https://docs.docker.com/get-docker/. Please note that Docker must run as a non-root user, you can set this up by following these instructions: https://docs.docker.com/engine/install/linux-postinstall/
+
+## 1. Initialize services
+
+Ocean.js uses off-chain services for metadata (Aquarius) and consuming datasets (Provider).
+
+We start by initializing the services. To do this, we clone the Barge repository and run it. This will run the current default versions of [Aquarius](https://github.com/oceanprotocol/aquarius), [Provider](https://github.com/oceanprotocol/provider), and [Ganache](https://github.com/trufflesuite/ganache) with [our contracts](https://github.com/oceanprotocol/contracts) deployed to it.
+
+```bash
+git clone https://github.com/oceanprotocol/barge.git
+cd barge/
+./start_ocean.sh --with-provider2 --no-dashboard
+```
+
+## 2. Create a new node.js project
+
+Start by creating a new Node.js project. Open a new terminal and enter the following commands:
+
+```bash
+mkdir marketplace-quickstart
+cd marketplace-quickstart
+npm init
+# Answer the questions in the command line prompt
+touch marketplace.ts
```

Comment:
If we are now advising the user to write this in typescript we need to add an extra step for creating a `tsconfig.json` file + we need to provide a command for compiling and running the code.